code/log/lib/Loot/Log/Internal.hs

```diff
@@ -104,24 +110,64 @@ pkgName full = extract . T.breakOnAll "-" . toText $ full
     extract ((n, _) : [_]) = n
     extract (_ : ns)       = extract ns
 
+-- | One may wish to use names derived from callstack
+-- or manually specified ones.
+data NameSelector
+    = CallstackName
+    | GivenName Name
+
+-- | Take a 'Name' according to 'NameSelector'.
+selectLogName :: HasCallStack => CallStack -> NameSelector -> Name
+selectLogName cs CallstackName = nameFromStack cs
+selectLogName _ (GivenName n)  = n
 
 -- | Logging capability.
 data Logging m = Logging
-    { _log :: Level -> Name -> Text -> m ()
+    { _log     :: Level -> Name -> Text -> m ()
+    , _logName :: m NameSelector
```

Comment:
How user is supposed to change it in runtime if he wants, for example, run a specific function with some other logger name?

Reply:
Having monad here is something `makeCap` forces me to do.
Added lens to modify logger name in last commit.